dcpy/connectors/edm/recipes.py

```diff
@@ -35,26 +38,60 @@ class VersionStrategy(str, Enum):
     bump_latest_release = "bump_latest_release"
 
 
-class Dataset(BaseModel, use_enum_values=True):
+class DatasetType(str, Enum):
+    pg_dump = "pg_dump"
+    csv = "csv"
+
+
+_dataset_extensions = {"pg_dump": "sql", "csv": "csv"}
```

Comment:
Like `dataset.value` wasn't typed as a string?

Reply:
mypy seems happy with this

```python
class DatasetType(str, Enum):
    pg_dump = "pg_dump"
    csv = "csv"

    @property
    def extension(self) -> str:
        match self:
            case DatasetType.pg_dump:
                return "sql"
            case DatasetType.csv:
                return "csv"
```

Reply:
And below 

```python
class Dataset(BaseModel, use_enum_values=True, extra="forbid"):
    name: str
    version: str | None = None
    version_env_var: str | None = None
    import_as: str | None = None
    file_type: DatasetType | None = None
    preprocessor: DataPreprocessor | None = None

    def is_resolved(self):
        return self.version is not None and self.version != "latest"

    @property
    def file_name(self) -> str:
        return (
            f"{self.name}.{self.file_type.extension}"
            if self.file_type is not None
            else ""
        )
```

Reply:
Interesting... lemme give it one more go. Thanks for investigating!

Reply:
Though I never actually tested it at runtime 🙂

Reply:
Ok, I was mis-remembering. It's a runtime error... It's something to do with assigning defaults when they're not specified in the recipe. I'll poke around with a few more minutes.

Reply:
mypy is also happy with

```python
class DatasetType(str, Enum):
    pg_dump = "pg_dump"
    csv = "csv"

    @property
    def extension(self) -> str:
        match self:
            case "pg_dump":
                return "sql"
            case "csv":
                return "csv"
            case _:
                return ""
```

Though at this point why use enum and match/case rather than a dict.

I'd say leave your code as is